node/core/av-store/src/lib.rs

```diff
@@ -53,374 +57,342 @@ mod columns {
-struct PoVPruningRecord {
-	candidate_hash: CandidateHash,
-	block_number: BlockNumber,
-	candidate_state: CandidateState,
-	prune_at: PruningDelay,
-}
+fn load_meta(
+	db: &Arc<dyn KeyValueDB>,
+	hash: &CandidateHash,
+) -> Result<Option<CandidateMeta>, Error> {
+	let key = (META_PREFIX, hash).encode();
 
-impl PartialEq for PoVPruningRecord {
-	fn eq(&self, other: &Self) -> bool {
-		self.candidate_hash == other.candidate_hash
-	}
+	query_inner(db, columns::META, &key)
 }
 
-impl Ord for PoVPruningRecord {
-	fn cmp(&self, other: &Self) -> Ordering {
-		if self.candidate_hash == other.candidate_hash {
-			return Ordering::Equal;
-		}
+fn write_meta(
+	tx: &mut DBTransaction,
+	hash: &CandidateHash,
+	meta: &CandidateMeta,
+) {
+	let key = (META_PREFIX, hash).encode();
 
-		self.prune_at.cmp(&other.prune_at)
-	}
+	tx.put_vec(columns::META, &key, meta.encode());
 }
 
-impl PartialOrd for PoVPruningRecord {
-	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-		Some(self.cmp(other))
-	}
+fn delete_meta(tx: &mut DBTransaction, hash: &CandidateHash) {
+	let key = (META_PREFIX, hash).encode();
+	tx.delete(columns::META, &key[..])
 }
 
-#[derive(Debug, Decode, Encode, Eq)]
-struct ChunkPruningRecord {
-	candidate_hash: CandidateHash,
+fn delete_unfinalized_height(
+	tx: &mut DBTransaction,
 	block_number: BlockNumber,
-	candidate_state: CandidateState,
-	chunk_index: u32,
-	prune_at: PruningDelay,
+) {
+	let prefix = (UNFINALIZED_PREFIX, BEBlockNumber(block_number)).encode();
+	tx.delete_prefix(columns::META, &prefix);
 }
 
-impl PartialEq for ChunkPruningRecord {
-	fn eq(&self, other: &Self) -> bool {
-		self.candidate_hash == other.candidate_hash &&
-			self.chunk_index == other.chunk_index
-	}
+fn delete_unfinalized_inclusion(
+	tx: &mut DBTransaction,
+	block_number: BlockNumber,
+	block_hash: &Hash,
+	candidate_hash: &CandidateHash,
+) {
+	let key = (
+		UNFINALIZED_PREFIX,
+		BEBlockNumber(block_number),
+		block_hash,
+		candidate_hash,
+	).encode();
+
+	tx.delete(columns::META, &key[..]);
 }
 
-impl Ord for ChunkPruningRecord {
-	fn cmp(&self, other: &Self) -> Ordering {
-		if self.candidate_hash == other.candidate_hash {
-			return Ordering::Equal;
-		}
+fn delete_pruning_key(tx: &mut DBTransaction, t: impl Into<BETimestamp>, h: &CandidateHash) {
+	let key = (PRUNE_BY_TIME_PREFIX, t.into(), h).encode();
+	tx.delete(columns::META, &key);
+}
 
-		self.prune_at.cmp(&other.prune_at)
-	}
+fn write_pruning_key(tx: &mut DBTransaction, t: impl Into<BETimestamp>, h: &CandidateHash) {
+	let t = t.into();
+	let key = (PRUNE_BY_TIME_PREFIX, t, h).encode();
+	tx.put(columns::META, &key, TOMBSTONE_VALUE);
 }
 
-impl PartialOrd for ChunkPruningRecord {
-	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-		Some(self.cmp(other))
-	}
+fn finalized_block_range(finalized: BlockNumber) -> (Vec<u8>, Vec<u8>) {
+	let start = UNFINALIZED_PREFIX.encode();
+	let end = (UNFINALIZED_PREFIX, BEBlockNumber(finalized + 1)).encode();
```

Comment:
Although it may seem self-evident but it would be nice to add a comment somewhere explaining that `be` is used to iterate in ascending order